### ISSUE TITLE
ALX: a program that prints its name, followed by a new line

### ISSUE THREAD

**Issue by ahmed149** [ahmed149/learning-programing-in-C#59]
#include <stdio.h>

/**
 * main - Prints the program name, followed by a new line.
 * @argc: The number of arguments supplied to the program.
 * @argv: An array of pointers to the arguments.
 *
 * Return: Always o.
 */
int main(int  argc, char const *argv[])
{
	(void)argc;
	printf("%s\n", argv[0]);

	return (0);
}

**Comment by ahmed149** [comment]
*)If you rename the program, it will print the new name, without having to compile it again
*)You should not remove the path before the name of the program
gcc -Wall -pedantic -Werror -Wextra -std=gnu89 0-whatsmyname.c -o mynameis
./mynameis 
mv mynameis mynewnameis
./mynewnameis

**Comment by ahmed149** [comment]
Sure, I'll explain without giving you code and be as specific as possible, just like you asked.

In the program you provided, there are a few things going on. Let me break it down for you:

1. `int main(int argc, char const *argv[])`:
   - `argc` stands for "argument count," and it tells you how many arguments (pieces of information) were given to the program when you run it.
   - `argv` stands for "argument vector," and it's an array of strings (text) that contains those arguments.

2. `const` before `char *argv[]`:
   - `const` is used here to indicate that the content of `argv` should not be modified inside the `main` function. It means that the program promises not to change the values in `argv`. This is a good practice because it helps prevent accidental modifications to the command line arguments.

3. `(void)argc;`:
   - This line is a way to tell the compiler that you are not going to use the `argc` variable in your program. It's done to avoid a "compiler warning" about an unused variable. So, it's just there to keep the compiler happy, and it doesn't do anything in the program.

4. `printf("%s\n", argv[0]);`:
   - This line prints the name of the program itself (the first argument in `argv`). The `%s` in `printf` is a placeholder for a string, and `argv[0]` contains the name of the program.

So, to summarize:
- `const` is used to make sure you don't accidentally change the command line arguments in your program.
- `(void)argc;` is used to tell the compiler that you won't use the `argc` variable.
- `printf("%s\n", argv[0]);` is used to print the name of the program.

I hope this explanation helps! If you have more questions or need further examples, feel free to ask.
